var/spack/repos/builtin/packages/libzip/package.py

```diff
@@ -51,9 +54,36 @@ def url_for_version(self, version):
         conditional("cmake", when="@1.4:"), conditional("autotools", when="@:1.3"), default="cmake"
     )
 
+    # Required dependencies
+    with when("build_system=cmake"):
+        variant("gnutls", default=True, description="Enable gnutls support")
+        variant("bzip2", default=True, description="Enable bzip2 support")
+        variant("lzma", default=True, description="Enable lzma support")
+        variant("openssl", default=True, description="Enable openssl support")
+        variant("zstd", default=True, description="Enable zstd support")
```

Comment:
Some initial questions:
`fmi4cpp/package.py` has: depends_on("libzip")
So `fmi4cpp` does not need all those variants enabled by default, correct?

I just wonder: As all those variants are new, how `libzip` was just fine all the time in spack not needing those other variants.

Question 2: Are those variants also supported by the older versions of libzip that have versions in the `spack` recipe?

Observation: all of these appear to be new, and they are now all enabled by default.

Normally, I'd think that bzip2, lzma (likely also and zstd) should be quite fine as espezially bzip2 and lzma are very common. I just wonder how libzip did without them.

At least lzma and bzip are small and fast to build and I guess the same for zstd.

That's different for openssl, gnutls and mbedtls, one might not like them and disable them.

Questions:
What is the benefit of enabling multiple of those in addition to the dedicated compression libraries?

What is their use in libzip?

Do you need all 3 of them enabled by default at the same time?

openssl is a very usual dependency of many spack packages, so it would be a default that would already be built more often than mbedtls or gnutls.

Would that be fine for you to disable all those variants by default (except may be `bzip2` and `lzma`, or even `zstd` as they are very common)?

Reply:
@bernhardkaindl I set the same defaults as in the cmake of libzip.

please the PR for libzip is actually here #47230 could you please that one instead. I made the mistake to merge the libzip branch to the develop branch on feelpp/spack ( I needed it for some other work but now it spreads through the other PR I made from feelpp/spack

Reply:
@bernhardkaindl all the variant are enabled by default by libzip so if they are not installed by spack, libzip will have a look on the host system and use that. That's what happened to me and it created havock in my apps at linking stage because of conflicting libs. the previous versions of libzip spack package was not very good from this point of view.

I checked up to version 1.7 of libzip and all the variants are enabled: 
https://github.com/nih-at/libzip/blob/v1.7.1/CMakeLists.txt#L10